app/gui/view/debug_scene/cloud-dashboard/src/projects_table.rs

```diff
@@ -0,0 +1,698 @@
+/// update the [`Model`] with network-fetched [`Project`]s data and re-render the [`ProjectsTable`]
+/// accordingly.
+///
+/// [`Project`]: ::enso_cloud_view::project::Project
+#[derive(Clone, Debug)]
+pub struct View {
+    frp:   Frp,
+    model: Model,
+}
+
+
+// === Internal `impl` ===
+
+impl View {
+    fn new(app: &Application) -> Self {
+        let frp = Frp::new();
+        let model = Model::new(app);
+        Self { frp, model }.init(app)
+    }
+
+    fn init(self, app: &Application) -> Self {
+        let frp = &self.frp;
+        let model = &self.model;
+        let root = &model.display_object;
+        let input = &frp.public.input;
+
+        self.init_projects_table_model_data_loading();
+        self.init_projects_table_grid_resizing();
+        self.init_projects_table_entries_models();
+        self.init_projects_table_grid();
+        self.init_projects_table_header();
+        self.init_event_tracing();
+
+        app.display.add_child(root);
+
+        populate_table_with_data(input.clone_ref());
+
+        self
+    }
+
+    fn init_projects_table_model_data_loading(&self) {
+        let frp = &self.frp;
+        let network = &frp.network;
+        let model = &self.model;
+        let input = &frp.public.input;
+
+        frp::extend! { network
+            eval input.set_projects((projects) model.set_projects(projects.clone_ref()));
+        }
+    }
+
+    fn init_projects_table_grid_resizing(&self) {
+        let network = &self.frp.network;
+        let model = &self.model;
+        let scene = &model.application.display.default_scene;
+        let projects_table = &model.projects_table;
+        let scroll_frp = projects_table.scroll_frp();
+
+        frp::extend! { network
+            scroll_frp.resize <+ scene.frp.shape.map(f!((shape) model.resize_grid_to_shape(shape)));
+            eval scene.frp.shape ((shape) model.reposition_grid_to_shape(shape));
+            projects_table.set_entries_size <+ scene.frp.shape.map(f!((shape) model.refit_entries_to_shape(shape)));
+        }
+    }
+
+    fn init_projects_table_entries_models(&self) {
+        let network = &self.frp.network;
+        let model = &self.model;
+        let projects_table = &model.projects_table;
+
+        frp::extend! { network
+            // We want to work with our `Position` struct rather than a coordinate pair, so convert.
+            needed_entries <- projects_table.model_for_entry_needed.map(|position| Position::from(*position));
+            // The first row is the header row, which is displayed with a different entry model.
+            needed_entries_body <- needed_entries.filter(|&position| position.row > 0);
+
+            projects_table.model_for_entry <+
+                needed_entries_body.filter_map(f!((position) model.model_for_entry(*position).map(|(position, entry_model)| {
+                    let (row, col) = position.into();
+                    (row, col, entry_model)
+                })));
```

Comment:
Not sure if I expressed it in the previous comment, but I think it's better to _not_filter out the requested for first row, but return the actual headers here as well.

Reply:
Yes you did express that and I somehow missed that comment in my review. Fixed now :)